src/x509.c

```diff
@@ -7067,8 +7067,10 @@ int wolfSSL_X509_REQ_print(WOLFSSL_BIO* bio, WOLFSSL_X509* x509)
             return WOLFSSL_FAILURE;
     }
 
-    /* print version of cert */
-    if (X509PrintVersion(bio, wolfSSL_X509_version(x509), 8)
+    /* print version of cert.  Note that we increment by 1 because for REQs,
+     * the value stored in x509->version is the actual value of the field; not
+     * the version. */
+    if (X509PrintVersion(bio, (int)wolfSSL_X509_REQ_get_version(x509) + 1, 8)
```

Comment:
Is this really the right way/place to adjust the version? The `wolfSSL_X509_get_version` and `wolfSSL_X509_set_version` do it differently.

Reply:
Yes, I think the reason this is correct is because this is specific for REQs; not certificates.

Reply:
Shouldn’t the +1 be inside the get function? Or is this the same way openssl does it?

Reply:
If the +1 was in the wolfSSL_X509_REQ_get_version() then it would return the wrong value.  This behaviour matches OpenSSL.